### PR TITLE
Issue #1548

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will address Issue #1548 

### Proposed changes

- add valid property strings
- add test for strings including the same strings as the enumerations

### CODE REVIEW

**Comment by phlptp** [comment]
@kdheepak  @nightlark  There were some issues with the formatting action, that is causing some issues.  This one needs to be merged next to resolve them.  Then we might have to rebase the others.

**Review by nightlark — APPROVED** [review]

File: src/helics/application_api/FederateInfo.cpp [hunk 1]
```diff
@@ -37,37 +38,64 @@ FederateInfo::FederateInfo(const std::string& args)
     loadInfoFromArgsIgnoreOutput(args);
 }
 
-static const std::map<std::string, int> propStringsTranslations{
+static const std::unordered_map<std::string, int> propStringsTranslations{
     {"period", helics_property_time_period},
+    {"timeperiod", helics_property_time_period},
+    {"timeperiod", helics_property_time_period},
```
Comment (kdheepak) [inline src/helics/application_api/FederateInfo.cpp:44]: Aren't these two lines the same?

```suggestion
    {"timeperiod", helics_property_time_period},
```

**Review by kdheepak — APPROVED** [review]

File: src/helics/application_api/FederateInfo.cpp [hunk 2]
```diff
@@ -37,37 +38,64 @@ FederateInfo::FederateInfo(const std::string& args)
...
+    {"timeinputdelay", helics_property_time_input_delay},
+    {"timeoutputdelay", helics_property_time_output_delay},
+    {"time_input_delay", helics_property_time_input_delay},
+    {"time_output_delay", helics_property_time_output_delay},
     {"loglevel", helics_property_int_log_level},
     {"log_level", helics_property_int_log_level},
-    {"maxiterations", helics_property_int_max_iterations},
     {"logLevel", helics_property_int_log_level},
+    {"intloglevel", helics_property_int_log_level},
+    {"intlog_level", helics_property_int_log_level},
```
Comment (kdheepak) [inline src/helics/application_api/FederateInfo.cpp:82]: Why are we adding `intlog_level` and not `int_log_level` when later we are adding `int_file_log_level`. 

I think we shouldn't have any `int{}` or `time{}` variant. Basically, I would like to only support users entering the attribute as we use it in the code but without the `helics_property_time_` or `helics_property_int_` prefix.

I think there should be just one and obvious way to use a property. For example, I think it is confusing for users whether `iterations` and `max_iterations` are the same thing or not

File: src/helics/application_api/FederateInfo.cpp (hunk 1 above)
Comment (phlptp) [reply src/helics/application_api/FederateInfo.cpp:44]: I think I meant to add an underscore

File: src/helics/application_api/FederateInfo.cpp (hunk 2 above)
Comment (phlptp) [reply src/helics/application_api/FederateInfo.cpp:82]: The "int" and "time" was what you were testing in the issue this is a based on.  Also in helics3 @trevorhardy  removed a lot of the variants so when I port this over, many will be removed.  I can remove them happily but then it won't technically be resolving the issue?

Comment (phlptp) [reply src/helics/application_api/FederateInfo.cpp]: @kdheepak   Do you think we should have the "time" and "int" variants like suggested in #1642 ?

**Comment by phlptp** [comment]
@kdheepak  I am going to merge this with the int and time strings.  We can discuss further if we want to remove them later I am fine either way.  But this is blocking some other stuff right now.

**Pull request merged**